packages/forgetti/runtime/index.ts

```diff
@@ -12,14 +12,20 @@ export interface Ref<T> {
 }
 export type RefHook = <T>(callback: T) => Ref<T>;
 
-export function $$cache(hook: RefHook, size: number): unknown[] {
+export function $$ref(hook: RefHook, size: number): unknown[] {
   const ref = hook<unknown[] | undefined>(undefined);
   if (!ref.current) {
     ref.current = new Array(size);
   }
   return ref.current;
 }
 
+export type MemoHook = <T>(callback: () => T, dependencies: unknown[]) => T;
+
+export function $$cache(hook: MemoHook, size: number): unknown[] {
+  return hook(() => new Array<unknown>(size), []);
```

Comment:
would it save anything if the array instance here was a singleton/constant?

Reply:
Do you mean the dependency array here? Hoisting an empty array should help save some GC pressure!

Reply:
The dependency array is hoisted in [`e770368` (#30)](https://github.com/lxsmnsyc/forgetti/pull/30/commits/e77036899c02bda7c7cdcac7eb925c79f9cd2d5a).